### ISSUE TITLE
Parser validation fails

### ISSUE THREAD

**Issue by milosterzic** [phpmyadmin/phpmyadmin#15931]
### Describe the bug

Executing `\<?php echo \'\1\'\; ?>\` ends up in infinite loop.

### To Reproduce

1. Select database
2. Go to SQL tab
3. Type `\<?php echo \'\1\'\; ?>\`
4. Click Go button

### Expected behavior

Printing error message that this code isn't regular MySQL code. 

### Server configuration

- Operating system: Ubuntu 16.04.1
- Web server: nginx/1.13.12
- Database version: libmysql - mysqlnd 5.0.12-dev
- PHP version: 7.2
- phpMyAdmin version: 5.1.0-dev

### Client configuration

- Browser: Chrome
- Operating system: Ubuntu 16.04

### Additional context

I researched it for few hours now and it's a bug. So when you submit `\<?php echo \'\1\'\; ?>\`, eventually app will end up in ImportSql class.

On line :178 it will start infinite loop, because $bq->query is not empty (it has malicious string), but $bq->extract(true) will always return empty string. Basically, at that point user needs to wait php's max_execution_time to expire and at that moment CPU works on 100%. Whole server is slowed down for that time period. My virtual machine needs 10-20 seconds to just open simple HTML page.

**Comment by rabeehrz** [comment]
I wasn't able to reproduce this issue. I'm using PHP 7.3.9. Is that why?

![](https://i.imgur.com/ZpzmJLf.png)

**Comment by milosterzic** [comment]
@rabeehrz 

Github parser removed some characters from string. 

This is malicious string, try:
`\<?php echo \'\1\'\; ?>`

**Comment by tabraiz961** [comment]
Reproduced on php 7.2.24
Is this a parsing issue for the BufferedQuery class extract function?

**Comment by williamdes** [comment]
@tabraiz961 I could not reproduce the issue in an unit test in this repository and in the sql-parser repository

If someone can succeed reproducing this issue in a unit test I would be very happy because it would help solve the issue

**Comment by tabraiz961** [comment]
Tested on sql parser, i could not reproduce this issue. 
To reproduce this issue using unittests, I will be running the malicious string in pma_bookmark.sql file.
When running unit tests locally the "templates/" directory does not access the root path instead it call the current working directory causing directory not found issue. Is there a debug switch that toggles for finding the correct path locally ?
![image](https://user-images.githubusercontent.com/5255703/74332717-32cf7780-4db8-11ea-874f-88ccfdbd6b7e.png)

**Comment by williamdes** [comment]
@tabraiz961 I used
```
    /**
     * Test for doImport
     *
     * @return void
     *
     * @group medium
     */
    public function testDoImportBadString()
    {
        //$sql_query_disabled will show the import SQL detail
        global $sql_query, $sql_query_disabled;
        $sql_query_disabled = false;
        $GLOBALS['import_type'] = 'query';
        $GLOBALS['import_file'] = 'none';
        $GLOBALS['import_text'] = '\\<?php echo \\\'\\1\\\'\\; ?>\\';
        //Mock DBI
        $dbi = $this->getMockBuilder('PhpMyAdmin\DatabaseInterface')
            ->disableOriginalConstructor()
            ->getMock();
        $GLOBALS['dbi'] = $dbi;

        //Test function called
        $this->object->doImport();

        $this->assertSame(
            "\<?php echo \'\1\'\; ?>\\",
            $sql_query
        );

        $this->assertEquals(
            true,
            $GLOBALS['finished']
        );
    }
```

But I coud not make it work as I wanted

**Comment by williamdes** [comment]
I tried the other day and still can confirm it makes the server go wild, sill not sure why

**Comment by williamdes** [comment]
The code gets stuck here (ImportSql class)
```php

        // Extracting remaining statements.
        while (! $error && ! $timeout_passed && ! empty($bq->query)) {
            $statement = $bq->extract(true);
            if (empty($statement)) {
                continue;
            }

            $this->import->runQuery($statement, $statement, $sql_data);
        }
```

**Comment by williamdes** [comment]
Hi @iifawzi 
Would you like to investigate about this one ?

**Comment by iifawzi** [comment]
Sure, I'd love to give it a try. @williamdes. I've made a fast test and it seems like that the server will always go wild whenever the query is ended with `\`. You can test the same case mentioned in the issue without the last backslash and it will throw the errors as expected. You can even test with any query that ends with `\` and it will go infinitely. I will try to figure out why tomorrow

**Comment by williamdes** [comment]
Fixed by 5a40ae64f24843bab8dfe299259671187b127e83 (sql-parser 5.5 upgrade)
